airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -0,0 +1,448 @@
+        self._start_date = self.str2datetime(start_date)
+
+    def _prepare_query(self, updated_after: datetime = None):
+        """some ZenDesk provides the field 'query' where we can send more details filter information"""
+        conds = [f"type:{self.entity_type[:-1]}"]
+        conds.append("created>%s" % self.datetime2str(self._start_date))
+        if updated_after:
+            conds.append("updated>%s" % self.datetime2str(updated_after))
+        return {"query": " ".join(conds)}
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        next_page = self._parse_next_page_number(response)
+        if next_page:
+            return {"next_page": next_page}
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        # Root of all responses of searching endpoints is 'results'
+        yield from response.json()["results"] or []
+
+    def path(self, *args, **kargs) -> str:
+        return "search.json"
+
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        updated_after = None
+        if stream_state and stream_state.get(self.cursor_field):
+            updated_after = self.str2datetime(stream_state[self.cursor_field])
+
+        # add the 'query' parameter
+        res = self._prepare_query(updated_after)
+        res.update(
+            {
+                "sort_by": "created_at",
+                "sort_order": "asc",
+                "size": self.state_checkpoint_interval,
+            }
+        )
+        if next_page_token:
+            res["page"] = next_page_token["next_page"]
+        return res
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        # try to save maximum value of a cursor field
+
+        return {
+            self.cursor_field: max(
+                (latest_record or {}).get(self.cursor_field, ""), (current_stream_state or {}).get(self.cursor_field, "")
+            )
+        }
+
+
+class IncrementalBasicEntityStream(IncrementalBasicSearchStream, ABC):
+    """basic stream for endpoints where an entity name can be used in a path value
+    https://<subdomain>.zendesk.com/api/v2/<entity_name>.json
+    """
+
+    # for generation of a path value and as rule as JSON root name of all response
+    entity_type: str = None
+
+    # for partial cases when JSON root name of responses is not equal a entity_type value
+    response_list_name: str = None
+
+    def path(self, *args, **kwargs) -> str:
+        return f"{self.entity_type}.json"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """returns data from API AS IS"""
+        yield from response.json()[self.response_list_name or self.entity_type] or []
```

Comment:
For me - it looks dangerous :) , maybe it is better to use like:

response.json().get(self.response_list_name or self.entity_type)